core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -95,13 +95,114 @@ oppia.directive('topNavigationBar', [function() {
         }
 
         $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
+        var currentWindowWidth = windowDimensionsService.getWidth();
+        $scope.navElementsVisibilityStatus = {
+          I18N_TOPNAV_DONATE: true,
+          I18N_TOPNAV_ABOUT: true,
+          I18N_CREATE_EXPLORATION_CREATE: true,
+          I18N_TOPNAV_LIBRARY: true
+        };
+
         windowDimensionsService.registerOnResizeHook(function() {
           $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
           $scope.$apply();
           // Close the sidebar, if necessary.
           SidebarStatusService.closeSidebar();
+
+          // If the window is resized larger, try displaying the hidden elements
+          if (currentWindowWidth < windowDimensionsService.getWidth()) {
+            $.each($scope.navElementsVisibilityStatus, function(e, visible) {
+              if (!visible) {
+                $scope.navElementsVisibilityStatus[e] = true;
+              }
+            });
+          }
+          currentWindowWidth = windowDimensionsService.getWidth();
+          oppiaDebouncer.debounce(truncateNavbar, 500)();
```

Comment:
Making those changes worked, thank you.

Should I add a commit to use Angular's debouncer in the library here or make a new issue and/or PR for that?

I'm not sure what would be best as far as moving it to TopNavigationBarDirective or leaving it in app.js. I had initially thought the factory pattern worked like the constructor pattern, i.e. each use would have its own private vars.

One solution I can do for leaving it in app.js (in case of other future uses) could be changing it to return a constructor function, like this: http://stackoverflow.com/q/18072683